devicetypes/sinope-technologies/va4200wz-va4200zb-sinope-valve.src/va4200wz-va4200zb-sinope-valve.groovy

```diff
@@ -286,6 +284,34 @@ private int get_LOG_TRACE() {
 	return 5
 }
 
+private def convertVoltToPercent(value) {
+    def NBDEPOINTS_DEG = 6;
+    def levelValue;
+    def levelsTable = [0, 20000, 40000, 60000, 80000, 100000];
+    def anglesTable = [30, 55, 56, 57, 58.5, 60];
+
+    if (value > anglesTable[NBDEPOINTS_DEG - 1]) { // if the value of the angle is greater than the maximum
+      value = anglesTable[NBDEPOINTS_DEG - 1]; // use the maximum value instead
+    }
+
+    def index = 1;
+    for (index; value > anglesTable[index]; index++) {
+    }
+
+    def ratioBetweenPointXandY = (levelsTable[index] - levelsTable[index - 1]) / (anglesTable[index] - anglesTable[index - 1]);
+    def angleToAdd = levelsTable[index] - (anglesTable[index] * ratioBetweenPointXandY);
+    def levelWithFactor = (ratioBetweenPointXandY * value) + angleToAdd;
+    levelValue = levelWithFactor / 1000;
+    def roundedLevelValue = Math.round(levelValue);
+    if(roundedLevelValue > 100){
```

Comment:
Please add space after ifs, space after a conditions, and space before else/else if-s in this file.